src/safir/dependencies/db_session.py

```diff
@@ -0,0 +1,119 @@
+"""Manage an async database session."""
+
+from typing import TYPE_CHECKING, AsyncIterator, Optional
+
+from sqlalchemy.ext.asyncio import AsyncEngine, async_scoped_session
+
+from ..database import create_async_session, create_database_engine
+
+# _IsolationLevel is defined in the SQLAlchemy type stubs as a long list of
+# Literal options, which is not type-compatible with str.  Use this hack to
+# use the correct list for type checking, but not for regular execution when
+# the type stubs will not be loaded.
+if TYPE_CHECKING:
+    from sqlalchemy.engine.create import _IsolationLevel
+else:
+    _IsolationLevel = str
+
+__all__ = ["DatabaseSessionDependency", "db_session_dependency"]
+
+
+class DatabaseSessionDependency:
+    """Manages an async per-request SQLAlchemy session.
+
+    Notes
+    -----
+    Creation of the database session factory has to be deferred until the
+    configuration has been loaded, which in turn is deferred until app
+    startup.  An app that uses this dependency must call:
+
+    .. code-block:: python
+
+       await db_session_dependency.initialize(database_url)
+
+    from its startup hook and:
+
+    .. code-block:: python
+
+       await db_session_dependency.aclose()
+
+    from its shutdown hook.
```

Comment:
For clarity, I tend not to use blocks within a sentence. Instead, you might want to introduce the `initialize` code sample by saying, "In the startup hook:` and introduce the `aclose` sample by saying "In the shutdown hook:"